engines/wasmtime/wasmtime.go

```diff
@@ -75,10 +79,26 @@ type (
 var _ = (wapc.Module)((*Module)(nil))
 var _ = (wapc.Instance)((*Instance)(nil))
 
-var engineInstance = engine{}
+type EngineOption func(e *engine)
 
-func Engine() wapc.Engine {
-	return &engineInstance
+func Engine(opts ...EngineOption) wapc.Engine {
+	e := engine{}
+	for _, opt := range opts {
+		opt(&e)
+	}
+	return &e
+}
+func WithDebug(b bool) EngineOption {
+	return func(e *engine) {
+		e.isDebug = b
+	}
+}
+
+func WithMetrics(maxGas uint64) EngineOption {
```

Comment:
isn't the word for something like this metering, not metrics? metrics usually is different, like perf metrics or error count.

Reply:
I'm changing these to WithMetering.